authority/tls.go

```diff
@@ -645,6 +645,85 @@ func (a *Authority) GetTLSCertificate() (*tls.Certificate, error) {
 	return &tlsCrt, nil
 }
 
+// GetTLSClientCertificate creates a new leaf certificate to be used for webhook
+// requests.
+func (a *Authority) GetTLSClientCertificate() (*tls.Certificate, error) {
+	// Generate default key.
+	priv, err := keyutil.GenerateDefaultKey()
+	if err != nil {
+		return nil, err
+	}
+	signer, ok := priv.(crypto.Signer)
+	if !ok {
+		return nil, errors.New("private key is not a crypto.Signer")
+
+	}
+
+	// prepare the sans: IPv6 DNS hostname representations are converted to their IP representation
```

Comment:
Does the client cert need SANs?

Reply:
This is an interesting question because it can send internal names to the server. Some SAN might be useful but not required.

If we end up using the same tls.Config as the server, we might be sending internal names. It really makes this easier, but I'm not sure if this is what we should do.

Reply:
We will use the same certificate, but we should not use the same `tls.Client`